src/Compilers/Core/Portable/SourceGeneration/Nodes/SourceOutputNode.cs

```diff
@@ -35,11 +35,11 @@ public SourceOutputNode(IIncrementalGeneratorNode<TInput> source, Action<SourceP
 
         public IncrementalGeneratorOutputKind Kind => _outputKind;
 
-        public NodeStateTable<TOutput> UpdateStateTable(DriverStateTable.Builder graphState, NodeStateTable<TOutput> previousTable, CancellationToken cancellationToken)
+        public NodeStateTable<TOutput> UpdateStateTable(DriverStateTable.Builder graphState, NodeStateTable<TOutput>? previousTable, CancellationToken cancellationToken)
         {
             string stepName = Kind == IncrementalGeneratorOutputKind.Source ? WellKnownGeneratorOutputs.SourceOutput : WellKnownGeneratorOutputs.ImplementationSourceOutput;
             var sourceTable = graphState.GetLatestStateTableForNode(_source);
-            if (sourceTable.IsCached)
+            if (sourceTable.IsCached && previousTable is not null)
```

Comment:
>previousTable is not null [](http://example.com/codeflow?start=40&length=25)

Are we testing this case? If it's not possible to hit, consider asserting that `previousTable` cannot be `null` when `sourceTable.IsCached`.

Reply:
Thanks. It's possible to hit this. Added a test that gets here with `sourceTable.IsCached == true` and `previousTable == null`.